File: frontend/src/lib/api/pricing.ts
import { apiClient } from './client'

export interface PricingPlan {
  id: string
  name: string
  price: number
  currency: string
  period: string
  description: string
  features: string[]
  popular: boolean
  active: boolean
  max_notebooks: number | null
  max_sources: number | null
  created: string
  updated: string
}

export interface Subscription {
  id: string
  user_id: string
  plan: string
  status: string
  start_date: string
  end_date: string | null
  auto_renew: boolean
  created: string
  updated: string
}

export interface SubscriptionStats {
  total_subscriptions: number
  active_subscriptions: number
  cancelled_subscriptions: number
}

export interface SubscribeRequest {
  plan_id: string
  user_id: string
}

/**
 * Get all active pricing plans
 */
export async function getPricingPlans(): Promise<PricingPlan[]> {
  const response = await apiClient.get('/pricing/plans')
  console.log(response,"hhhh");
  
  return response.data
}

/**
 * Get a specific pricing plan by ID
 */
export async function getPricingPlan(planId: string): Promise<PricingPlan> {
  const response = await apiClient.get(`/pricing/plans/${planId}`)
  return response.data
}

/**
 * Subscribe to a pricing plan
 */
export async function subscribeToPlan(data: SubscribeRequest): Promise<Subscription> {
  const response = await apiClient.post('/pricing/subscribe', data)
  return response.data
}

/**
 * Get user's active subscription
 */
export async function getUserSubscription(userId: string): Promise<Subscription | null> {
  try {
    const response = await apiClient.get(`/pricing/subscription/${userId}`)
    return response.data
  } catch (error: any) {
    // Return null if no subscription found (404 is expected)
    if (error?.response?.status === 404) {
      return null
    }
    throw error
  }
}

/**
 * Cancel a subscription
 */
export async function cancelSubscription(subscriptionId: string): Promise<{ message: string }> {
  const response = await apiClient.post(`/pricing/subscription/${subscriptionId}/cancel`)
  return response.data
}

/**
 * Get subscription statistics (admin only)
 */
export async function getSubscriptionStats(): Promise<SubscriptionStats> {
  const response = await apiClient.get('/pricing/stats')
  return response.data
}

File: open_notebook/domain/pricing.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field


class PricingPlan(BaseModel):
    """Domain model for pricing plan"""
    id: Optional[str] = None
    name: str
    price: float
    currency: str = "USD"
    period: str = "month"
    description: str
    features: List[str]
    popular: bool = False
    active: bool = True
    max_notebooks: Optional[int] = None
    max_sources: Optional[int] = None
    created: Optional[datetime] = None
    updated: Optional[datetime] = None


class UserSubscription(BaseModel):
    """Domain model for user subscription"""
    id: Optional[str] = None
    user_id: str
    plan: str  # Record ID of pricing_plan
    status: str = "active"  # active, cancelled, expired
    start_date: datetime
    end_date: Optional[datetime] = None
    auto_renew: bool = True
    created: Optional[datetime] = None
    updated: Optional[datetime] = None
